### PR TITLE
"Failed Unusual log in for (O365, AWS, GWS)

### PR DESCRIPTION
**Pull request by roaaattalla**

### CODE REVIEW

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -2547,6 +2547,57 @@ action.cyences_notable_event_action.system_compromised_drilldown = `cs_o365` sou
...
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_o365` sourcetype="o365:management:activity" _index_earliest=-61m@m _index_latest=-1m@m  Workload=AzureActiveDirectory Operation=UserLoginFailed  `cs_public_ips(ClientIP)` \
+| fields _time sourcetype user ClientIP Id LogonError user_type authentication_method app ApplicationId \
+| append \
+    [| search `cs_o365` sourcetype=o365:graph:api source="AuditLogs.SignIns" earliest=-2h@m latest=now  status.errorCode=0 `cs_public_ips(ipAddress)` \
+    | fields _time sourcetype userPrincipalName ipAddress id LogonError appDisplayName status.failureReason clientAppUsed conditionalAccessStatus isInteractive deviceDetail.* org_country org_region org_city \
```
Comment (roaaattalla) [inline cyences_app_for_splunk/default/savedsearches.conf:2574]: For (status.failReason) when I run the query it doesn't show data .. I tried several things to make it work but it doesn't show any data so can you please check the query maybe you can recognize if I'm missing something

**Review by hardikhdholariya — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -2547,6 +2547,57 @@ action.cyences_notable_event_action.system_compromised_drilldown = `cs_o365` sou
...
+    | rename userPrincipalName as user, ipAddress as ClientIP, id as Id, status.failureReason as failureReason] \
+| eval user = lower(user) \
+| stats values(*) as *, max(_time) as Last_Failed_Login by Id, user, ClientIP \
+| search sourcetype="o365:management:activity" \
+| fields - Id, sourcetype, lat, lon, deviceDetail.deviceId, \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:2579]: remove unwanted command from the end of line.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -5611,6 +5706,48 @@ action.cyences_notable_event_action.system_compromised_drilldown = `cs_gws_login
...
+\
+Data Collection - Google login activity reports data from Splunk Add-on for Google Workspace (https://splunkbase.splunk.com/app/5556) 
+dispatch.earliest_time = -61m@m
+dispatch.latest_time = 1m@m
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_gws_login_events` "event.name"="login_failure" `cs_public_ips(ipAddress)` \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:5729]: Add **_index_earliest=-61m@m _index_latest=-1m@m** in the search

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -5611,6 +5706,48 @@ action.cyences_notable_event_action.system_compromised_drilldown = `cs_gws_login
...
+| fields _time user ipAddress login_challenge_method app \
+| eval user = lower(user) \
+| stats count, values(*) as *, max(_time) as Last_Failed_Login by user, ipAddress \
+| iplocation ipAddress \
+| `cs_current_week_login_count(gws:reports:login)` \
+| `cs_country_login_percentage` \
+| fillnull Country, Region, City value="-" \
+| `cs_user_privilege_mapping(user)` \
+| eval cyences_severity = case(percentage_login_from_country==0 or or is_privileged_user=="Yes", "critical", true(), "medium") \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:5738]: remove **or**, it's typed twice in sequence.

**Review by hardikhdholariya — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 5]
```diff
@@ -5611,6 +5706,48 @@ action.cyences_notable_event_action.system_compromised_drilldown = `cs_gws_login
...
+| where count > 5 \
+| table user, ipAddress, count, cyences_severity, Country, Region, City, percentage_login_from_country, usual_login_location, Last_Failed_Login, login_challenge_method, reason, is_privileged_user \
+| `cs_human_readable_time_format(Last_Failed_Login)` \
+| `cs_gws_failed_login_from_unusual_country_filter` 
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_failed_login_from_unusual_country_filter
+action.cyences_notable_event_action.contributing_events = `cs_gws_login_events` "event.name"="login_failure" `cs_public_ips(ipAddress)` | iplocation ipAddress
+action.cyences_notable_event_action.system_compromised_search = | stats count by user
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_gws_login_events` "event.name"="login_failure"`cs_public_ips(ipAddress)` user=$row.user$ | iplocation ipAddress
```
[the remark this answers is not in the document]
Comment (roaaattalla) [reply cyences_app_for_splunk/default/savedsearches.conf:5747]: Done no errors

**Review by hardikhdholariya — APPROVED** [review]

**Pull request merged**